node/storage.go

```diff
@@ -0,0 +1,184 @@
+package node
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/storetheindex/store"
+	"github.com/ipfs/go-cid"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+)
+
+var _ store.Storage = &nodeStorage{}
+
+// nodeStorage combines a cache and a persistent storage
+type nodeStorage struct {
+	primary    store.Storage
+	persistent store.PersistentStorage
+}
+
+// NewStorage creates new nodeStorage from a primary and persistent storages
+func NewStorage(primary store.Storage, persistent store.PersistentStorage) *nodeStorage {
+	return &nodeStorage{primary, persistent}
+}
+
+// Get retrieves IndexEntries for a CID
+func (ns *nodeStorage) Get(c cid.Cid) ([]store.IndexEntry, bool, error) {
+	if ns.primary != nil {
+		// Check if CID in primary storage
+		v, found, err := ns.primary.Get(c)
+
+		if ns.persistent != nil && (!found || err != nil) {
+			v, found, err = ns.persistent.Get(c)
+			// TODO: What about adding a CacheStorage interface that includes
+			// putEntries(cid, []IndexEntry) function
+			// so we don't need to loop through IndexEntry to move from
+			// one storage to another?
+			if found && err == nil {
```

Comment:
```suggestion
			if found && ns.primary != nil {
```
Need to check that there is a primary storage.

Reply:
This is already checked at the [beginning of this block](https://github.com/filecoin-project/storetheindex/pull/17/files#diff-059931e6619ad69718bd9674f123a7445e26ef0badd0a68118ff23bdc5b48b7fR26), right? We can't reach this code if there is no primary storage. In any case, the `err` check is redundant.